$NetBSD: patch-makefile,v 1.4 2021/10/14 19:12:56 rhialto Exp $

Exclude -flto for gcc 4.1.3 (used in NetBSD/vax 6)
Exclude -flto for clang on NetBSD
Exclude -flto for gcc 4.4.7 (Used in DragonFly 2.13 - 3.1)
NO_LTO=1 is in effect for gcc and clang, except Darwin.
Make -O2 conditional so can be overridden by environment
Fix png16 detection

--- makefile.orig	2021-10-07 06:38:08.000000000 +0000
+++ makefile
@@ -298,7 +298,7 @@ ifeq (${WIN32},)  #*nix Environments (&&
       $(shell git log -1 --pretty="SIM_GIT_COMMIT_ID %H$(GIT_EXTRA_FILES)%nSIM_GIT_COMMIT_TIME $(isodate)" >.git-commit-id)
     endif
   endif
-  LTO_EXCLUDE_VERSIONS = 
+  LTO_EXCLUDE_VERSIONS = 4.1.3 4.4.7 4.8.3 4.8.4 4.8.5 5.3.0 5.4.0
   PCAPLIB = pcap
   ifeq (agcc,$(findstring agcc,${GCC})) # Android target build?
     OS_CCDEFS = -D_GNU_SOURCE -DSIM_ASYNCH_IO 
@@ -598,9 +598,9 @@ ifeq (${WIN32},)  #*nix Environments (&&
     OS_CCDEFS += -DHAVE_UTIME
   endif
   ifneq (,$(call find_include,png))
-    ifneq (,$(call find_lib,png))
+    ifneq (,$(call find_lib,png16))
       OS_CCDEFS += -DHAVE_LIBPNG
-      OS_LDFLAGS += -lpng
+      OS_LDFLAGS += -lpng16
       $(info using libpng: $(call find_lib,png) $(call find_include,png))
       ifneq (,$(call find_include,zlib))
         ifneq (,$(call find_lib,z))
@@ -1185,11 +1185,11 @@ ifneq (,$(UNSUPPORTED_BUILD))
 endif
 ifneq ($(DEBUG),)
   CFLAGS_G = -g -ggdb -g3
-  CFLAGS_O = -O0
+  CFLAGS_O ?= -O0
   BUILD_FEATURES = - debugging support
 else
   ifneq (,$(findstring clang,$(COMPILER_NAME))$(findstring LLVM,$(COMPILER_NAME)))
-    CFLAGS_O = -O2 -fno-strict-overflow
+    CFLAGS_O ?= -O2 -fno-strict-overflow
     GCC_OPTIMIZERS_CMD = ${GCC} --help
     NO_LTO = 1
   else
@@ -1197,7 +1197,7 @@ else
     ifeq (Darwin,$(OSTYPE))
       CFLAGS_O += -O4 -flto -fwhole-program
     else
-      CFLAGS_O := -O2
+      CFLAGS_O ?= -O2
     endif
   endif
   LDFLAGS_O = 
